### PR TITLE
48800 remove UserIdentity attributes

### PR DESCRIPTION
**Pull request by bramleyjl**
## Summary

- This PR removes sourcing from UserIdentity for the following attributes:
```
  - common_name
  - suffix
  - birls_id
  - participant_id
  - address
  - postal_code
  - phone
  - cerner_id
  - cerner_facility_ids
  - vha_facility_ids
  - vha_facility_hash
```
The long-term plan is to reduce `UserIdentity`'s role in vets-api to just handling a user's primary identifying traits & ids strictly related to authentication & login (CSP ids & MPI ICN). Once a non-LoA1 user has been created & MPI has been queried for their attributes the User model should be referring to MPI as the source of truth for as many attributes as possible.
- *(Which team do you work for, does your team own the maintainence of this component?)*
Most of this work is within the scope of Identity but other services that depend on specific user attributes are impacted.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#48800
- [*Link to ticket created in va.gov-team repo*](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48800)

## Testing done

- *Describe the steps required to verify your changes are working as expected*
  - Update a user's MPI profile mockdata & authenticate, you will see the updated values reflected on the User model & User serializer
- *Describe the tests completed and the results*
  - Affected unit tests updated
  - MPI sourcing of attributes covered in `user_spec`
  - Authentication testing with SiS & SSOe & confirmation via breakpoints that affected traits are still accessible

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
- The User & UserIdentity models have been heavily impacted
- BGS service (`lib/bgs/service.rb`) has been changed to only pass a user's email (still sourced from UserIdentity, so always available) instead of first attempting a user's `common_name` (now no longer a UserIdentity attribute). From my investigation of this service it seemed that invocations of this service with `common_name` only existed in spec files where it was actually being passed an email under that attribute, so I removed the unused option entirely. 
- Swagger schema for `/v0/user` / the User model serializer has had the `zip` attribute that maps to a User's `postal_code` now can have a `null` value for LoA1 users due to them being unable to source that attribute from MPI
- Numerous spec files & factories have been impacted to repair tests that depended on UserIdentity-sourced attributes. Of particular note is the `mpi_attr_sourcing` trait in the User factory; this is the beginning of a "default" inclusion of MPI data for all LoA3 user factories as the User model relies more heavily on MPI for attribute sourcing.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

## Requested Feedback
- Confirmation from developers more familiar with the BGS service that the change to remove a user's common name for a BGS `external_key` will not have adverse impacts.

### CODE REVIEW

File: lib/bgs/service.rb [hunk 1]
```diff
@@ -273,11 +273,9 @@ def create_note(claim_id, note_text)
     private
 
     def service
-      external_key = @user.common_name || @user.email
-
       @service ||= BGS::Services.new(
         external_uid: @user.icn || @user.uuid,
-        external_key: external_key
+        external_key: @user.email
```
Comment (bosawt) [inline lib/bgs/service.rb:278]: Can you also make this change in `bgs/base_service.rb`? They have a similar way to set `external_key`, but we should prob make these consistent

**Review by bosawt — CHANGES_REQUESTED** [review]
I tested with Sign in Service and SSOe, authentication worked for both. Additionally I went on a rails console and tried `user.common_name`, `user.suffix`, `user.birls_id`, `user.participant_id`, `user.address`, `user.postal_code`, all of these worked. `user.phone` didn't work, but `user.home_phone` did, which is analogous. Additionally, I wasn't able to find any place in the code base that still referred to `user.phone`, everything appears to use `user.home_phone`, so I think that's good.

Only had a single quick comment, and then I'll approve

**Review by bosawt — DISMISSED** [review]
Looks good!

**Review by rjohnson2011 — DISMISSED** [review]

**Review by ajmagdub — COMMENTED** [review]
Tried to test the changes against the VAOS-VAOS Service JWT creation. From this PRs View Deployment selected 'Sign In' then 'ID.me' which brought up a 502 Bad Gateway error page. Cannot verify this has not broken any VAOS processes.

**Review by rjohnson2011 — APPROVED** [review]

**Review by bosawt — APPROVED** [review]

**Pull request merged**